backend/app/api/api_v1/endpoints/expectation.py

```diff
@@ -97,172 +66,98 @@ def list_expectations(
         suggested: Optional[bool] = None,
         enabled: Optional[bool] = True,
         asc: Optional[bool] = False,
+        repository: ExpectationRepository = Depends(get_expectation_repository),
 ):
-    direction = "asc" if asc else "desc"
-    sort_by_key: str = "expectation_type"
-
-    query = {"query": {"bool": {"must": []}}, "sort": [{sort_by_key: direction}]}
-
-    query["query"]["bool"]["must"].append({"match": {"enabled": enabled}})
-
-    if suggested is not None:
-        query["query"]["bool"]["must"].append({"match": {"suggested": suggested}})
-
-    if datasource_id is not None:
-        query["query"]["bool"]["must"].append({"match": {"datasource_id": datasource_id}})
-
-    if dataset_id is not None:
-        query["query"]["bool"]["must"].append({"match": {"dataset_id": dataset_id}})
-
-    try:
-        if include_history:
-            body = [
-                {"index": settings.EXPECTATION_INDEX},
-                {**{"size": 1000}, **query},
-                {"index": settings.VALIDATION_INDEX},
-                validation.validations_query_body(datasource_id, dataset_id),
-            ]
-
-            results = client.msearch(body=body)
-
-            expectations = results["responses"][0]["hits"]["hits"]
-            validations = results["responses"][1]["hits"]["hits"]
-
-            return JSONResponse(
-                status_code=status.HTTP_200_OK,
-                content=zip_expectations_and_validations(expectations, validations)
-            )
+    expectations = repository.query_by_filter(
+        datasource_id=datasource_id,
+        dataset_id=dataset_id,
+        suggested=suggested,
+        enabled=enabled,
+        asc=asc,
+    )
 
+    if include_history:
         results = client.search(
-            index=settings.EXPECTATION_INDEX,
-            size=1000,
-            body=query
-        )["hits"]["hits"]
-    except RequestError as ex:
-        print(ex)
-        raise HTTPException(
-            status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
-            detail=f"invalid sort_by_key"
-        )
-
-    result_response = []
-    for result in results:
-        source = result["_source"]
-        expectation_type = source["expectation_type"]
-        source["kwargs"] = json.loads(source["kwargs"])
-        expectation = exp.type_map[expectation_type](**source)
-        source["documentation"] = expectation.documentation()
```

Comment:
We need to include documentation in the response. It's currently missing.

Reply:
Ok!